File: BackEnd/src/main/java/tfg/backend/models/Asignatura.java
package tfg.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.Collection;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import lombok.Data;

@Entity
@Data
public class Asignatura {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String descripcion;
    private String curso;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "asignatura_id")
    @JsonIgnore
    private Collection<Section> sections;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    private Usuario profesor;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "asignatura_id")
    @JsonIgnore
    private Collection<Tutoring> tutorings;

    @ManyToMany
    @JsonIgnore
    private Collection<Matricula> matriculas;
}

File: FrontEnd/src/app/models/asignatura.ts
import { Usuario } from './usuario';
import { Materiales } from './materiales';
import { Section } from './section';
import { Matricula } from './matricula';

export class Asignatura {
    id: string;
    nombre: string;
    descripcion: string;
    curso: string;
    profesor: Usuario;
    materiales: Materiales[];
    matriculas: Matricula[];
    sections: Section[];

    constructor() {
        this.id = null;
        this.nombre = null;
        this.nombre = null;
        this.descripcion = null;
        this.curso = null;
        this.profesor = new Usuario();
    }
}
